pkg/maps/devicemap/devicemap.go

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package devicemap
+
+import (
+	"fmt"
+	"log/slog"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/cilium/cilium/pkg/ebpf"
+	"github.com/cilium/cilium/pkg/mac"
+)
+
+const (
+	MapName    = "cilium_device_map"
+	MaxEntries = 256
+)
+
+// Map provides access to the eBPF map node.
+type Map interface {
+	// Lookup returns the device map object associated with the provided
+	// ifIndex.
+	Lookup(ifIndex uint32) (DeviceValue, error)
+
+	// Update inserts or updates the device map object associated with the provided
+	// ifIndex, mac and l3.
+	Update(ifIndex uint32, mac mac.Uint64MAC, l3 uint8) error
+
+	// Delete deletes the device map object associated with the provided
+	// ifIndex.
+	Delete(ifIndex uint32) error
+
+	// IterateWithCallback iterates through all the keys/values of a device map,
+	// passing each key/value pair to the cb callback.
+	IterateWithCallback(cb DeviceIterateCallback) error
+}
+
+type deviceMap struct {
+	bpfMap *ebpf.Map
```

Comment:
We want to actually get rid of the `pkg/ebpf` package. Could you refactor to use the `pkg/bpf` package instead, which should have all features needed. https://github.com/cilium/cilium/issues/33128